lib/eslint.js

```diff
@@ -994,21 +994,32 @@ module.exports = (function() {
      */
     api.markVariableAsUsed = function(name) {
         var scope = this.getScope(),
-            variables,
-            i,
-            len;
-
-        do {
-            variables = scope.variables;
-            for (i = 0, len = variables.length; i < len; i++) {
-                if (variables[i].name === name) {
-                    variables[i].eslintUsed = true;
-                    return true;
+            found = false;
+
+        function markVariableAsUsedInScope(currentScope) {
+            var variables,
+                i,
+                len;
+            do {
+                variables = currentScope.variables;
+                for (i = 0, len = variables.length; i < len; i++) {
+                    if (variables[i].name === name) {
+                        variables[i].eslintUsed = true;
+                        return true;
+                    }
                 }
-            }
-        } while ( (scope = scope.upper) );
+            } while ( (currentScope = currentScope.upper) );
+
+            return false;
+        }
+
+        found = markVariableAsUsedInScope(scope);
+
+        if (!found && scope.childScopes.length) {
```

Comment:
This is overly aggressive because there's only one time when we want to check the first child. You really want to check if `scope.type` is `"global"` and only do this next step if that's true.